### ISSUE TITLE
Old-style script works different from BrainScriptNetworkBuilder

### ISSUE THREAD

**Issue by bulatGH** [Microsoft/CNTK#1515]
I rewrote my old network using BrainScriptNetworkBuilder, but results are always Nan.
What could be the problem?

train_Net = {

	action = "train"

    BrainScriptNetworkBuilder = [
	
        tsize=27
		labelDim = 2
		imageShape = 3 * tsize:tsize:1
		
		cMap1 = 20
		cMap2 = 50
		cMap3 = 50
		h1Dim = 100
		
		model = features1 => {
            
			feature_x = Slice(0, tsize, features1)
			feature_y = Slice(tsize, 2 * tsize, features1)
			feature_z = Slice(tsize, 3 * tsize, features1)
			
			conv1_x   = ConvolutionalLayer {cMap1, (5:5), pad = false, activation=ReLU}(feature_x)
			conv1_y   = ConvolutionalLayer {cMap1, (5:5), pad = false, activation=ReLU}(feature_y)
			conv1_z   = ConvolutionalLayer {cMap1, (5:5), pad = false, activation=ReLU}(feature_z)
			conv1_xd  = Dropout (conv1_x)
			conv1_yd  = Dropout (conv1_y)
			conv1_zd  = Dropout (conv1_z)
			pool1_x   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv1_xd)
			pool1_y   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv1_yd)
			pool1_z   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv1_zd)
			
			conv2_x   = ConvolutionalLayer {cMap2, (5:5), pad = false, activation=ReLU}(pool1_x)
			conv2_y   = ConvolutionalLayer {cMap2, (5:5), pad = false, activation=ReLU}(pool1_y)
			conv2_z   = ConvolutionalLayer {cMap2, (5:5), pad = false, activation=ReLU}(pool1_z)
			conv2_xd  = Dropout (conv2_x)
			conv2_yd  = Dropout (conv2_y)
			conv2_zd  = Dropout (conv2_z)
			pool2_x   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv2_xd)
			pool2_y   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv2_yd)
			pool2_z   = MaxPoolingLayer {(2:2:2), stride=(2:2:2)}(conv2_zd)
			
			conv3_x   = ConvolutionalLayer {cMap3, (3:3), pad = false, activation=ReLU}(pool2_x)
			conv3_y   = ConvolutionalLayer {cMap3, (3:3), pad = false, activation=ReLU}(pool2_y)
			conv3_z   = ConvolutionalLayer {cMap3, (3:3), pad = false, activation=ReLU}(pool2_z)
			
			conv_Final = RowStack(conv3_x : conv3_y : conv3_z);
			
			h_final = DenseLayer {h1Dim, activation=ReLU} (conv_Final)
			o_final = LinearLayer {labelDim, init='normal', initValueScale=0.01} (h_final)
        }.o_final
		
	# inputs
	features = Input {imageShape}
        labels   = Input {labelDim}

	ol = model (features)
	ce       = CrossEntropyWithSoftmax(labels, ol)
        errs     = ClassificationError(labels, ol)
		
        # declare special nodes
        featureNodes    = (features)
        labelNodes      = (labels)
        criterionNodes  = (ce)
        evaluationNodes = (errs)
        outputNodes     = (ol)
    ]

    SGD = {
        learningRatesPerSample=0.0001
		minibatchSize=250
		momentumPerMB = 0.9
		epochSize=0
		maxEpochs=10
		dropoutRate=0.5
    }

    # The reader is a composite reader that uses the ImageReader to read images
    # and the CNTKTextFormatReader to read the regression ground truth labels.
    # It does so by defining an array of deserializers (using {...} : {...} below)
    # and assigning the inputs as defined in the network above (cf. features and regrLabels).

	reader = {
        readerType = "CNTKTextFormatReader"
        # See DataSets/MNIST/README.md for details on getting the data (Train-28x28_cntk_text.txt).
        #file = "$DataDir$/$LandmarkType$/$Cases$.txt"
		file = "$DataDir$\$LandmarkType$\$Cases$.txt"
        randomize = true
        keepDataInMemory = true
		input = {
            features = { dim = 2187 ; format = "dense" }
            labels =   { dim = 2 ; format = "dense" }
        }
    }
}

**Comment by bulatGH** [comment]
Ok, I found out what was the problem - I forgot batch norm
